src/other/metadata.jl

```diff
@@ -0,0 +1,1072 @@
+end
+
+emptycolmetadata!(df::DataFrame, col::Symbol) =
+    emptycolmetadata!(df, Int(index(df)[col]))
+emptycolmetadata!(df::DataFrame, col::ColumnIndex) =
+    emptycolmetadata!(df, Int(index(df)[col]))
+
+function emptycolmetadata!(df::DataFrame)
+    setfield!(df, :colmetadata, nothing)
+    return df
+end
+
+function emptycolmetadata!(x::DataFrameRows, col::Int)
+    emptycolmetadata!(parent(x), col)
+    return x
+end
+
+function emptycolmetadata!(x::DataFrameColumns, col::Int)
+    emptycolmetadata!(parent(x), col)
+    return x
+end
+
+function emptycolmetadata!(x::DataFrameRows, col::Symbol)
+    emptycolmetadata!(parent(x), col)
+    return x
+end
+
+function emptycolmetadata!(x::DataFrameColumns, col::Symbol)
+    emptycolmetadata!(parent(x), col)
+    return x
+end
+
+function emptycolmetadata!(dfr::DataFrameRows, col::ColumnIndex)
+    emptycolmetadata!(parent(dfr), col)
+    return dfr
+end
+
+function emptycolmetadata!(dfc::DataFrameColumns, col::ColumnIndex)
+    emptycolmetadata!(parent(dfc), col)
+    return dfc
+end
+
+function emptycolmetadata!(x::Union{DataFrameRows, DataFrameColumns})
+    emptycolmetadata!(parent(x))
+    return x
+end
+
+function emptycolmetadata!(x::DataFrameRow, col::Int)
+    col_name = _names(x)[col]
+    df = parent(x)
+    idx = index(df)[col_name]
+
+    for key in colmetadatakeys(df, idx)
+        _, s = colmetadata(df, idx, key, style=true)
+        s == :note && deletecolmetadata!(df, idx, key)
+    end
+    return x
+end
+
+function emptycolmetadata!(x::SubDataFrame, col::Int)
+    col_name = _names(x)[col]
+    df = parent(x)
+    idx = index(df)[col_name]
+
+    for key in colmetadatakeys(df, idx)
+        _, s = colmetadata(df, idx, key, style=true)
+        s == :note && deletecolmetadata!(df, idx, key)
+    end
+    return x
+end
+
+emptycolmetadata!(x::DataFrameRow, col::Symbol) =
+    emptycolmetadata!(x, Int(index(x)[col]))
+emptycolmetadata!(x::SubDataFrame, col::Symbol) =
+    emptycolmetadata!(x, Int(index(x)[col]))
+emptycolmetadata!(x::DataFrameRow, col::ColumnIndex) =
+    emptycolmetadata!(x, Int(index(x)[col]))
+emptycolmetadata!(x::SubDataFrame, col::ColumnIndex) =
+    emptycolmetadata!(x, Int(index(x)[col]))
+
+function emptycolmetadata!(x::DataFrameRow)
+    for i in 1:length(x)
+        emptycolmetadata!(x, i)
+    end
+    return x
+end
+
+function emptycolmetadata!(x::SubDataFrame)
+    for i in 1:ncol(x)
+        emptycolmetadata!(x, i)
+    end
+    return x
+end
+
+### Internal utility functions for metadata handling
+
+# copy table level :note metadata from src to dst
+# discarding previous metadata contents of dst
+function _copy_df_note_metadata!(dst::DataFrame, src)
+    emptymetadata!(dst)
+    for key in metadatakeys(src)
+        val, style = metadata(src, key, style=true)
+        style === :note && metadata!(dst, key, val, style=:note)
+    end
+    return nothing
+end
+
+# copy column level :note metadata from src to dst from column src_col to dst_col
```

Comment:
`src_col` not as in general it could potentially be anything.
`dst_col` can have a more specific type, but I prefer it to be caught in `emptycolmetadata!(dst, dst_col)` which is a first function called since it is public.